File: 1791-find-center-of-star-graph/1791-find-center-of-star-graph.java
class Solution {
    public int findCenter(int[][] edges) {
         Set<Integer> components = new HashSet<>();
        Map<Integer, LinkedList<Integer>> map = new HashMap<>();
        for (int[] edge : edges) {
            int u = edge[0];
            int v = edge[1];
            components.add(u);
            components.add(v);
            map.putIfAbsent(u, new LinkedList<>());
            map.putIfAbsent(v, new LinkedList<>());
            map.get(u).add(v);
            map.get(v).add(u);
        }
        int center = -1;
        for (int component : components) {
            if (map.get(component).size() == components.size() - 1) {
                center = component;
                break;
            }
        }
        return center;
    }
}

File: 1791-find-center-of-star-graph/1791-find-center-of-star-graph.py
class Solution:
    def findCenter(self, edges: List[List[int]]) -> int:
        n = len(edges) + 1
        d = defaultdict(int)
        for x, y in edges:
            d[x] += 1
            d[y] += 1
        for k, v in d.items():
            if v == n - 1:
                return k
        